### PR TITLE
Add an example with SSR caching.

### PR DESCRIPTION
**Pull request by arunoda**
This is pretty interesting :)

### CODE REVIEW

File: examples/ssr-caching/server.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    .then((html) => {
+      // Let's cache this page
+      console.log(`CACHE MISS: ${req.url}`)
+      ssrCache.set(req.url, html)
+
+      res.send(html)
+    })
+    .catch((err) => {
+      console.error(err.stack)
+      res.status(500).send('Internal Error')
```
Comment (nkzawa) [inline examples/ssr-caching/server.js]: I think we can use `app.renderError` or `app.renderErrorToHTML` here

Comment (arunoda) [inline examples/ssr-caching/server.js]: Done!

Comment (jaredpalmer) [inline examples/ssr-caching/server.js]: This is awesome

**Comment by rauchg** [comment]
Really impressive @arunoda

**Comment by sedubois** [comment]
Doesn't this go against the idea of separating concerns? Shouldn't caching be done by an external service?

**Comment by arunoda** [comment]
@sedubois You could use a separate service like squid proxy. But this is for caching HTML (SSR), usually you need to bake some custom logic into when to invalidate the cache and so on.

Doing that is kind a hard. Due to that reason, most of the caching servers/services do not cache HTML by default.

**Comment by sedubois** [comment]
Could you please explain what was the reasoning behind choosing an LRU cache vs e.g LFU or other cache eviction strategy? Do you have other good cache NPM packages to recommend?

**Comment by sedubois** [comment]
Also @arunoda what about the SSR pages asked by next-prefetcher? At the moment those requests bypass the cache completely (e.g `/_next/-/pages/`, etc), although if there are any `<Link prefetch ...>` in the app, it will be the main load on the server.

**Comment by sedubois** [comment]
Oh I keep forgetting, next-prefetcher doesn't do a pre-render, it just downloads the code.

**Pull request merged**